ladybug/stat.py

```diff
@@ -191,13 +176,58 @@ def _import_data(self):
                 r"Monthly Statistics for Wind Speed[\s\S]*Daily Avg(.*)")
             for direction in self._wind_dir_names:
                 re_string = r"Monthly Wind Direction %[\s\S]*" + direction + r"\s(.*)"
-                self._monthly_wind_dirs.append(self._regex_parse(re_string))
+                dirs = self._regex_parse(re_string)
+                if dirs != []:
+                    self._monthly_wind_dirs.append(dirs)
+
+            # check to see if all monthly data is there
+            if self._monthly_db_range_50 != [] and self._monthly_wb_range_50 != [] \
+                    and self._monthly_wind != [] and self._monthly_wind_dirs != [] \
+                    and self._stand_press_at_elev is not None:
+                        self._monthly_found = True
         finally:
             statwin.close()
 
+    def _regex_check(self, regex_str, search_space, numbr=True):
+        matches = re.compile(regex_str).findall(search_space)
+        if len(matches) > 0:
+            if numbr is True:
```

Comment:
what if you try to cast to number and if it failed just return it as a string? Then you can remove the number argument. I'm thinking if there will be a case that one wants a numeric value not to be returned as a number.

Reply:
Ok.  I guess that this is a place where try / except is ok.  It gets a bit complicated for lists of numbers where one of the numbers is "N" indicating that there's no value for one of the items in the list.  But I think I can still write it to deal with this.